java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -0,0 +1,189 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.bytecode.se;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.util.Printer;
+import org.sonar.java.bytecode.cfg.BytecodeCFGBuilder;
+import org.sonar.java.bytecode.loader.SquidClassLoader;
+import org.sonar.java.se.ExplodedGraph;
+import org.sonar.java.se.ProgramPoint;
+import org.sonar.java.se.ProgramState;
+import org.sonar.java.se.constraint.ConstraintManager;
+import org.sonar.java.se.symbolicvalues.SymbolicValue;
+import org.sonar.java.se.xproc.BehaviorCache;
+import org.sonar.java.se.xproc.MethodBehavior;
+import org.sonar.plugins.java.api.semantic.Symbol;
+
+import java.util.Collections;
+import java.util.Deque;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.Set;
+
+public class BytecodeEGWalker {
+
+  private static final int MAX_EXEC_PROGRAM_POINT = 2;
+  private static final int MAX_STEPS = 16_000;
+  private final BehaviorCache behaviorCache;
+
+  private ExplodedGraph explodedGraph;
+
+  @VisibleForTesting
+  Deque<ExplodedGraph.Node> workList;
+  ExplodedGraph.Node node;
+  ProgramPoint programPosition;
+  ProgramState programState;
+  int steps;
+  private Set<ExplodedGraph.Node> endOfExecutionPath;
+  private ConstraintManager constraintManager;
+  MethodBehavior methodBehavior;
+
+  BytecodeEGWalker(BehaviorCache behaviorCache){
+    this.behaviorCache = behaviorCache;
+  }
+
+  public MethodBehavior getMethodBehavior(Symbol.MethodSymbol symbol, SquidClassLoader classLoader) {
+    methodBehavior = behaviorCache.methodBehaviorForSymbol(symbol);
+    if(!methodBehavior.isComplete()) {
+      execute(symbol, classLoader);
+      methodBehavior.completed();
+    }
+    return methodBehavior;
+  }
+
+  private void execute(Symbol.MethodSymbol symbol, SquidClassLoader classLoader) {
+    explodedGraph = new ExplodedGraph();
+    constraintManager = new ConstraintManager();
+    programState = ProgramState.EMPTY_STATE;
+    workList = new LinkedList<>();
+    endOfExecutionPath = new LinkedHashSet<>();
+    steps = 0;
+    BytecodeCFGBuilder.BytecodeCFG bytecodeCFG = BytecodeCFGBuilder.buildCFG(symbol, classLoader);
+    bytecodeCFG.blocks.forEach(b -> System.out.println(b.printBlock()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)